plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -194,6 +221,60 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
                 Optional.ofNullable(nameMappingJson));
     }
 
+    @Override
+    public boolean isSupportedVersionType(ConnectorSession session, SchemaTableName tableName, PointerType pointerType, io.trino.spi.type.Type versioning)
+    {
+        switch (pointerType) {
+            case TEMPORAL:
+                return versioning instanceof TimestampWithTimeZoneType || versioning instanceof TimestampType;
+            case TARGET_ID:
+                return versioning == BIGINT;
+        }
+        throw new TrinoException(NOT_SUPPORTED, "Version type " + versioning + " is not supported");
+    }
+
+    private Optional<Long> getSnapshotIdFromVersion(Table table, Optional<ConnectorTableVersion> versionOptional)
+    {
+        Optional<Long> snapshotId = Optional.empty();
+        if (versionOptional.isPresent()) {
+            ConnectorTableVersion version = versionOptional.get();
+
+            switch (version.getPointerType()) {
+                case TEMPORAL:
+                    long epochMillis;
```

Comment:
add `verify` versionType is either `TS` or `TSWTZ`

Reply:
My understanding is that `isSupportedVersionType` already does the check, is checking again here still needed?

Reply:
I would still add it as a sanity check. So we see obvious failures if we change one place and not the other.

Reply:
updated